File: leetCode/remove-duplicate-Medium/remove-duplicate-medium.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
    count = 0
    limit = 0
    let varLimite = nums[0]

    for (let i = 0; i < nums.length; i++) {
        if (varLimite != nums[i]) {
            limit = 0
            varLimite = nums[i]
        }
        if (limit < 2) {
            limit++
            nums[count] = nums[i]
            count++
        }
    }

    return count
};

// test case 1
console.log(removeDuplicates([1, 1, 1, 2, 2, 3]));

// test case 2
console.log(removeDuplicates([0, 0, 1, 1, 1, 1, 2, 3, 3]));
//expected output  [0,0,1,1,2,2,3,3,4]

File: leetCode/remove-duplicate-Medium/remove-duplicate-medium.go
package main

import "fmt"

func removeDuplicates(nums []int) int {
	var count int
	var limit int
	varLimit := nums[0]

	for i := 0; i < len(nums); i++ {
		if nums[i] != varLimit {
			limit = 0
			varLimit = nums[i]
		}

		if limit < 2 {
			nums[count] = nums[i]
			count++
			limit++
		}
	}

	return count
}

func main() {
	nums := []int{1, 1, 1, 2, 2, 3}
	fmt.Println(removeDuplicates(nums))
}
